ast/Trees.h

```diff
@@ -87,26 +91,26 @@ class TreePtr {
 
         auto maskedPtr = reinterpret_cast<long long>(expr) & PTR_MASK;
 
-        return reinterpret_cast<void *>(maskedPtr | val);
+        return maskedPtr | val;
     }
 
     explicit TreePtr(Tag tag, void *expr) : ptr(tagPtr(tag, expr)) {}
 
     static void deleteTagged(Tag tag, void *ptr) noexcept;
 
     // A version of release that doesn't mask the tag bits
-    void *releaseTagged() noexcept {
-        auto *saved = ptr;
-        ptr = nullptr;
+    long long releaseTagged() noexcept {
```

Comment:
Why are some `long long` and some `tagged_storage`?

Reply:
I was just about to push a commit to fix this, thanks!